st2common/st2common/router.py

```diff
@@ -531,7 +531,12 @@ def __init__(self, **entries):
 
             try:
                 validator = CustomValidator(response_spec['schema'], resolver=self.spec_resolver)
-                validator.validate(resp.json)
+
+                response_type = response_spec['schema'].get('type', 'json')
```

Comment:
@enykeev I needed to make change since our router code didn't really correctly support non JSON responses.

It always called ``resp.json`` which would fail on non-json responses.